devicetypes/stelpro/stelpro-ki-zigbee-thermostat.src/stelpro-ki-zigbee-thermostat.groovy

```diff
@@ -418,7 +418,8 @@ def validateOperatingStateBugfix(map) {
 	if (state.rawSetpoint != null && state.rawTemp != null) {
 		def oldVal = map.value
 
-		if (state.rawSetpoint <= state.rawTemp) {
+		if (state.rawSetpoint <= state.rawTemp ||
+			(device.currentValue("thermostatMode") && device.currentValue("thermostatMode") == "off")) {
```

Comment:
The `(device.currentValue("thermostatMode")` check shouldn't be necessary since if it doesn't have a current value it won't be equal to "off".

Reply:
No, but wouldn't trying to do `null == "off"` result in a NPE?

Reply:
No, you can safely check for equality against null. You can verify this in the Groovy console: https://groovyconsole.appspot.com/

Reply:
Alright, updated that!